### DISCUSSION TITLE
Help Attaching Remote Database

### DISCUSSION THREAD

**Discussion by Admolly** [duckdb/duckdb#7781] [Q&A]
The blog post for the 0.7.0 Release mentioned that it was now possible to [attach remote databases](https://duckdb.org/2023/02/13/announcing-duckdb-070.html#multi-database-support), but the [documentation](https://duckdb.org/docs/sql/statements/attach) itself doesn't indicate how to accomplish this.  I tried:

- `ATTACH DATABASE 'https://github.com/<account_name>/<repo_name>/blob/main/<db_filename>'`
- `ATTACH DATABASE 'https://<bucket_name>.s3.eu-central-1.amazonaws.com/<db_filename>'`

both to no avail.  If someone has managed to make this work please share your insights!

**Comment by Tishj** [comment]
https://github.com/duckdb/duckdb/blob/master/test/sql/attach/attach_httpfs.test

This shows a test for this functionality
Are you sure you installed and loaded the HTTPFS extension?

**Reply by Admolly** [reply]
Here is my code:

```
con.execute(
    f"""
    INSTALL httpfs;
    LOAD httpfs;
    SET s3_region='eu-central-1';
    SET s3_access_key_id='{key_id}';
    SET s3_secret_access_key='{secret_key}';
    """
)
```

`con.execute(f"ATTACH 'https://github.com/{user}/{repo}/blob/main/{filename}'")`
`con.execute(f"ATTACH DATABASE 'https://{bucket}.s3.eu-central-1.amazonaws.com/{filename}'")`

Both of these throw the following error:

`InternalException: INTERNAL Error: State was not defined in this HTTP File Handle`

This is from a Jupyter Notebook environment on Windows 10 using the DuckDB 0.8.0 Python library.

The Github I'm not sure since it's a private repo and I have not provided authentication, but I don't see how to do that either.  The S3 I have authenticated so I'm not sure of the issue.

**Reply by Tishj** [reply]
Yea that's definitely not intended, as a rule of thumb INTERNAL error is not something a user should ever see.
This specific error has been seen before however, similar to #7748

**Reply by lhoestq** [reply]
I opened an issue here: https://github.com/duckdb/duckdb/issues/7998
